### PR TITLE
Create ClayTable

### PR DESCRIPTION
**Pull request by carloslancha**
Developed with clay#develop

Missing features:
- Sticker in main variant. Waiting for @pat270 to provide the markup.
- Horizontal scrolling when too many columns with fixed first column. Waiting for UX (@victorvalle) to define it and @pat270 to provide the markup.
- Dropdown out of the limits of the overvlow hidden container. We need to think about this.
- Test for focus feature is not working properly.

### CODE REVIEW

File: packages/clay-table/package.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    "compile": "babel -d lib/ src/ -s --ignore src/__tests__",
+    "prepublish": "npm run soy && npm run compile",
+    "soy": "metalsoy --soyDeps '../../node_modules/clay-+(button|checkbox|dropdown|icon|label|link|progress-bar|radio)/src/**/*.soy'"
+  },
+  "keywords": [
+    "clay",
+    "metal"
+  ],
+  "dependencies": {
+    "clay-button": "1.0.0-alpha.8",
```
Comment (matuzalemsteles) [inline packages/clay-table/package.json]: hey @carloslancha, Upgrade to the latest versions of Clay to not have some problems in compiling and jest.

Comment (carloslancha) [reply packages/clay-table/package.json]: Thx @matuzalemsteles I didn't realize versions changed while developing the component ;)

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: packages/clay-table/src/ClayTable.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+import templates from './ClayTable.soy.js';
+
+/**
+ * Metal ClayTable component.
+ */
+class ClayTable extends Component {
+	/**
+	 * @inheritDoc
+	 * Handles document click in order to hide menu.	 * Handles document click in order to hide menu.	 */
```
Comment (matuzalemsteles) [inline packages/clay-table/src/ClayTable.js]: Just organize. 🙂

Comment (carloslancha) [reply packages/clay-table/src/ClayTable.js]: D'oh!

**Comment by matuzalemsteles** [comment]
Awesome working @carloslancha. Merged, thanks!

**Pull request merged**